pkg/scheduler/schedule_one.go

```diff
@@ -130,42 +130,43 @@ func (sched *Scheduler) schedulingCycle(
 	if err != nil {
+
+		if err == ErrNoNodesAvailable {
+			status := framework.NewStatus(framework.UnschedulableAndUnresolvable).WithError(err)
+			return ScheduleResult{nominatingInfo: clearNominatedNode}, podInfo, status
+		}
+
+		fitError, ok := err.(*framework.FitError)
+		if !ok {
+			klog.ErrorS(err, "Error selecting node for pod", "pod", klog.KObj(pod))
+			return ScheduleResult{nominatingInfo: clearNominatedNode}, podInfo, framework.AsStatus(err)
+		}
+
 		// SchedulePod() may have failed because the pod would not fit on any host, so we try to
 		// preempt, with the expectation that the next time the pod is tried for scheduling it
 		// will fit due to the preemption. It is also possible that a different pod will schedule
 		// into the resources that were preempted, but this is harmless.
-		var (
-			nominatingInfo *framework.NominatingInfo
-			status         *framework.Status
-		)
-		if fitError, ok := err.(*framework.FitError); ok {
-			if !fwk.HasPostFilterPlugins() {
-				klog.V(3).InfoS("No PostFilter plugins are registered, so no preemption will be performed")
-			} else {
-				// Run PostFilter plugins to try to make the pod schedulable in a future scheduling cycle.
-				result, status := fwk.RunPostFilterPlugins(ctx, state, pod, fitError.Diagnosis.NodeToStatusMap)
-				if status.Code() == framework.Error {
-					klog.ErrorS(nil, "Status after running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", status)
-				} else {
-					msg := status.Message()
-					fitError.Diagnosis.PostFilterMsg = msg
-					klog.V(5).InfoS("Status after running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", msg)
-				}
-				if result != nil {
-					nominatingInfo = result.NominatingInfo
-				}
-			}
-			status = framework.NewStatus(framework.Unschedulable).WithError(err)
-		} else if err == ErrNoNodesAvailable {
-			nominatingInfo = clearNominatedNode
-			status = framework.NewStatus(framework.UnschedulableAndUnresolvable).WithError(err)
+		var nominatingInfo *framework.NominatingInfo
+
+		if !fwk.HasPostFilterPlugins() {
+			klog.V(3).InfoS("No PostFilter plugins are registered, so no preemption will be performed")
+			return ScheduleResult{}, podInfo, framework.NewStatus(framework.Unschedulable).WithError(err)
+		}
+
+		// Run PostFilter plugins to attempt to make the pod schedulable in a future scheduling cycle.
+		result, status := fwk.RunPostFilterPlugins(ctx, state, pod, fitError.Diagnosis.NodeToStatusMap)
+		if status.Code() == framework.Error {
+			klog.ErrorS(nil, "Status after running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", status)
 		} else {
-			klog.ErrorS(err, "Error selecting node for pod", "pod", klog.KObj(pod))
-			nominatingInfo = clearNominatedNode
-			status = framework.AsStatus(err)
+			msg := status.Message()
+			fitError.Diagnosis.PostFilterMsg = msg
+			klog.V(5).InfoS("Status after running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", msg)
```

Comment:
when https://github.com/kubernetes/kubernetes/pull/114456 merges, we will get the error in the reasons and will be returned as part of status.Message(), so we can change this if/else to 

```
msg := status.Message()
fitError.Diagnosis.PostFilterMsg = msg
if status.Code() == framework.Error {
  klog.ErrorS(nil, "After running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", msg)
} else {
  klog.V(5).InfoS("After running PostFilter plugins for pod", "pod", klog.KObj(pod), "status", msg)
}
```

Reply:
Always record the PostFilter messages whatever the status, make sense, updated.